### ISSUE TITLE
Procedure do not work with comments

### ISSUE THREAD

**Issue by cccs-jc** [apache/iceberg#4289]
this works
```
spark.sql("""
CALL catalog_name.system.remove_orphan_files(table => 'db.sample', dry_run => true)
""")
```

But this fails
```
spark.sql("""
/* comment */
CALL catalog_name.system.remove_orphan_files(table => 'db.sample', dry_run => true)
""")
```

Tools like dbt add comments to the statements that are executed. The comments are useful for auditing.

**Comment by kbendick** [comment]
Hi @cccs-jc! Thanks for reporting this.

Can you please let me know which Iceberg (and Spark) version you have tried this with (just your normal one would be fine, or the latest one you've tried). Also, can you provide me an example (redacted) comment that's more representative of the kind DBT has?

I see some comment handling code in our `IcebergSqlExtensions.g4` file which generates the antlr grammar, but some comments have meaning (SQL hints) and so having a closer example to what's really generated would be helpful while I take a look.

Thank you!

**Comment by RussellSpitzer** [comment]
I think we'd have to modify our grammer, since our parser uses it's own grammar which doesn't have comments. That or finally copy the grammar from the relevant version of spark and start using that instead of our own custom extension. This may be the time now that we separated by version

**Comment by cccs-jc** [comment]
dbt issues comments like this before it renders the procedure calls


/* {"app": "dbt", "dbt_version": "1.0.1", "profile_name": "profile1", "target_name": "dev", "node_id": "model.profile1.stg_users"} */

right now I modified my dbt-spark adapter to strip this out but I would rather keep the comments for auditing

**Comment by kbendick** [comment]
I think I have found a solution that seems to work for `CALL` statements. The issue is similar to this one, for making the iceberg statements more recognizable: https://github.com/apache/iceberg/pull/2729/files

I will open a draft PR shortly.

**Comment by kbendick** [comment]
I matched the admittedly wild regex used there against this test query statement. https://regex101.com/r/9cs1yV/1

Fortunately, this function doesn't modify the query at all, just how the query is determined to be an Iceberg query or not.

We still delegate to the _existing_ antlr parser for processing the query, and only strip comments out when we're checking if this is Iceberg specific DDL. Hence there's no concerns about accidentally dropping SQL hints, etc.

**Comment by kbendick** [comment]
This has been merged in for Spark 3.2. There are open PRs for Spark 3.0 and 3.1 as well. I'm going to close this issue now.

This should be released as part of Iceberg 0.14.0 (or the next non-patch release).

**Comment by kbendick** [comment]
@cccs-jc Please open a new issue if there are still problems, and reference this issue (also feel free to ping me in case I miss it) 👍
